packages/getting-started/src/browser/getting-started-contribution.ts

```diff
@@ -52,19 +78,49 @@ export class GettingStartedContribution extends AbstractViewContribution<Getting
     }
 
     async onStart(app: FrontendApplication): Promise<void> {
-        this.stateService.reachedState('ready').then(() => {
-            const editors = this.shell.widgets.filter((widget): widget is NavigatableWidget => NavigatableWidget.is(widget));
-            if (editors.length === 0) {
-                this.preferenceService.ready.then(() => {
-                    const showWelcomePage: boolean = this.preferenceService.get('welcome.alwaysShowWelcomePage', true);
-                    if (showWelcomePage) {
+        this.stateService.reachedState('ready').then(async () => {
+            if (this.editorManager.all.length === 0) {
+                await this.preferenceService.ready;
+                const startupEditor = this.preferenceService.get('welcome.startupEditor');
+                switch (startupEditor) {
+                    case 'welcomePage':
                         this.openView({ reveal: true, activate: true });
-                    }
-                });
+                        break;
+                    case 'welcomePageInEmptyWorkbench':
+                        if (!this.workspaceService.opened) {
+                            this.openView({ reveal: true, activate: true });
+                        }
+                        break;
+                    case 'newUntitledFile':
+                        this.commandRegistry.executeCommand(CommonCommands.NEW_UNTITLED_FILE.id);
```

Comment:
Close but I believe this is the wrong command, I mentioned `NEW_UNTITLED_TEXT_FILE` before :) It creates the file without the prompt for which type.